releasenotes/notes/update-instruction-alignment-passes-ef0f20d4f89f95f3.yaml

```diff
@@ -1,31 +1,165 @@
+    reworked transpiler workflow for scheduling. This new pass will insert dynamical decoupling
+    sequences into the circuit per any scheduling and alignment analysis that occured in earlier
+    passes.
+  - |
+    Introduced a new transpiler pass workflow for building :class:`~.PassManager` objects
+    for scheduling :class:`~.QuantumCircuit` objects in the transpiler. In the new
+    workflow scheduling and alignment passes are all :class:`~.AnalysisPass` objects that
+    only updates the property set of the pass manager, in which new property set
+    ``node_start_time`` is created there to perform these operation on nodes
+    scheduled on the absolute time. This new workflow is both more efficient and can correct
+    for additional timing constraints exposed by a backend.
+
+    Previously, the pass chain would have been implemented as ``scheduling -> alignment``
+    which were both transform passes thus there were multiple :class:`~.DAGCircuit`
+    instances recreated during each pass. In addition, scheduling occured in each pass
+    to obtain instruction start time. Now the required pass chain becomes
+    ``scheduling -> alignment -> padding`` where the :class:`~.DAGCircuit` update only
+    occurs at the end with the ``padding`` pass.
+
+    For those who are creating custom :class:`~.PassManager` objects that involve
+    circuit scheduling you will need to adjust your :class:`~.PassManager`
+    to include insert one of the :class:`~.BasePadding` passes (currently
+    either :class:`~.PadDelay` or :class:`~.DynamicalDecoupling` can be used)
+    at the end of the scheduling pass chain. Without the padding pass the scheduling
+    passes will not be reflected in the output circuit of the :meth:`~.PassManager.run`
+    method of your custom :class:`~.PassManager`.
+
+    For example, if you were previously building your :class:`~.PassManager`
+    with something like::
+
+        from qiskit.transpiler import PassManager
+        from qiskit.transpiler.passes import TimeUnitConversion, ALAPSchedule, ValidatePulseGates, AlignMeasures
+
+        pm = PassManager()
+        scheduling = [
+            ALAPSchedule(instruction_durations), PadDelay()),
+            ValidatePulseGates(granularity=timing_constraints.granularity, min_length=timing_constraints.min_length),
+            AlignMeasures(alignment=timing_constraints.acquire_alignment),
+        ]
+        pm.append(scheduling)
+
+    you can instead use::
+
+        from qiskit.transpiler import PassManager
+        from qiskit.transpiler.passes import TimeUnitConversion, ALAPScheduleAnalysis, ValidatePulseGates, AlignMeasures, PadDelay
+
+        pm = PassManager()
+        scheduling = [
+            ALAPScheduleAnalysis(instruction_durations), PadDelay()),
+            InstructionDurationCheck(acquire_alignment=timing_constraints.acquire_alignment, pulse_alignment=timing_constraints.pulse_alignment),
```

Comment:
This example is bit problematic because `InstructionDurationCheck` updates property set with `reschedule_required` and following `ConstrainedReschedule` is conditionally added to the pass manager, so `InstructionDurationCheck` do nothing here. 

This mechanism is introduced because we don't need to perform reschedule unless the circuit potentially violates the constraints. This is because `scheduling_method` is turned off by default in the `transpile`, thus directly adding rescheduler pass will always raise TranspileError for non-scheduled circuit. 

In this particular example, you added `ALAPScheduleAnalysis` anyways (in the preset pms this is conditionally added), thus `InstructionDurationCheck` is not necessary.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7835/commits/bae9f47aaa3ef990884f3de4a23ad439a0f59c52